pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBase.java

```diff
@@ -80,6 +91,11 @@ protected ConsumerBase(PulsarClientImpl client, String topic, ConsumerConfigurat
         this.pendingReceives = Queues.newConcurrentLinkedQueue();
         this.schema = schema;
         this.interceptors = interceptors;
+        this.batchReceivePolicy = conf.getBatchReceivePolicy();
+        this.pendingBatchReceives = Queues.newConcurrentLinkedQueue();
```

Comment:
Can you delay the creation of this queue util the first `batchReceive` is called? I am try to reduce creating the queue if the consumer is not using `batchReceive`.

Reply:
fix it